### ISSUE TITLE
Question about GAN training.

### ISSUE THREAD

**Issue by ForeverFancy** [VITA-Group/TransGAN#45]
Hi, thanks for the work. Follow the training code in [functions.py](https://github.com/VITA-Group/TransGAN/blob/master/functions.py#L154), it seems that you did not freeze D when training G. When running `dis_optimizer.step()`, the gradient from G training will also be used to update D's parameters. So I wonder whether if I missed something, or it was a bug here. Thanks a lot!

**Comment by yueruchen** [comment]
normally `dis_optimizer.zero_grad()` will erase the remaining gradient before updating D. Since we update G 1 time and D 4 times, I think 3/4 part of D updating is correct here. But I'm not sure whether the first updating of D will be affected, I might need to do some experiments to examine it.

**Comment by ForeverFancy** [comment]
OK, thanks.
